libs/shared/contentful/src/lib/contentful.stripe-mapper.ts

```diff
@@ -0,0 +1,234 @@
+import { Stripe } from 'stripe';
+import {
+  ContentfulManager,
+  OfferingCommonContentResult,
+  PurchaseDetailsTransformed,
+} from '@fxa/shared/contentful';
+import { StripeMetadataKeys } from './types';
+
+/**
+ *  TypeGuard to ensure product on plan is a Product Object
+ */
+const isProductObject = (
+  product: Stripe.Plan['product']
+): product is Stripe.Product => {
+  if (typeof product !== 'string' && product && !product.deleted) {
+    return true;
+  } else {
+    return false;
+  }
+};
```

Comment:
This breaks from the typings ADR but I assume that's because it has to be compatible with the old code?

Reply:
Additionally, any reason this isn't a private method within the class?

Reply:
Yup, this is working with legacy code so its necessary atm. Ha good question; will add it too the class.